clang/lib/CIR/CodeGen/CIRGenTypes.h

```diff
@@ -61,6 +64,13 @@ class CIRGenTypes {
 
   /// Convert a Clang type into a mlir::Type.
   mlir::Type convertType(clang::QualType type);
+
+  /// Convert type T into an mlir::Type. This differs from convertType in that
+  /// it is used to convert to the memory representation for a type. For
+  /// example, the scalar representation for bool is i1, but the memory
+  /// representation is usually i8 or i32, depending on the target.
```

Comment:
When is the memory type used in ClangIR? Given the target-dependence, it seems like we'd at least want this deferred until the lowering phase.

Reply:
`convertTypeForMem` is not currently used in the upstreamed code.  Earlier in this change when I was still sorting out exactly what code needed to be upstreamed, I thought it would be needed, so I included it here.  It is small enough and self contained enough and it completes the upstreaming of all the `convertType` functions that I decided to leave it in even though it isn't used yet.